common/v2/services/Store/Asset/helpers.ts

```diff
@@ -1,19 +1,15 @@
-import { readAll } from '../Cache';
-import { Asset, Network, StoreAsset } from 'v2/types';
+import { Asset, ExtendedAsset, Network, StoreAsset } from 'v2/types';
 import { generateUUID } from 'v2/utils';
 import { DEFAULT_ASSET_DECIMAL } from 'v2/config';
 
-export const getAllAssets = () => {
-  return readAll('assets')();
-};
-
-export const getAssetByTicker = (symbol: string): Asset | undefined => {
-  const assets: Asset[] = getAllAssets();
+export const getAssetByTicker = (assets: Asset[]) => (symbol: string): Asset | undefined => {
   return assets.find(asset => asset.ticker.toLowerCase() === symbol.toLowerCase());
 };
 
-export const getNewDefaultAssetTemplateByNetwork = (network: Network): Asset => {
-  const baseAssetOfNetwork: Asset | undefined = getAssetByUUID(network.baseAsset);
+export const getNewDefaultAssetTemplateByNetwork = (assets: Asset[]) => (
+  network: Network
+): Asset => {
+  const baseAssetOfNetwork: Asset | undefined = getAssetByUUID(assets)(network.baseAsset);
   if (!baseAssetOfNetwork) {
     return {
       uuid: generateUUID(),
```

Comment:
Removing the seedCache file exposes an issue with this function that results in base assets being created here having non-deterministic uuids.

Reply:
Good catch. Done!